### PR TITLE
Ensures early assets init when expected during unit tests, 

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

We can use this to catch errors in testing that'll pop up on live, without eating the 20 second init cost

## Why It's Good For The Game
moth got mad at me, needed immediate fixing

### CODE REVIEW

File: code/modules/asset_cache/asset_list.dm [hunk 1]
```diff
@@ -139,6 +139,13 @@ GLOBAL_LIST_EMPTY(asset_datums)
 	/// Defaults to false so we can process this stuff nicely
 	var/load_immediately = FALSE
 
+#ifdef DO_NOT_DEFER_ASSETS
+/datum/asset/spritesheet/New()
+	// ALWAYS load now
+	load_immediately = TRUE
```
Comment (Mothblocks) [inline code/modules/asset_cache/asset_list.dm:145]: Better to `#ifdef` the var declaration

Comment (Mothblocks) [reply code/modules/asset_cache/asset_list.dm:145]: Like:

```dm
#ifdef DO_NOT_DEFER_ASSETS
    var/load_immediately = TRUE
#else
    var/load_immediately = FALSE
#endif
```

Comment (LemonInTheDark) [reply code/modules/asset_cache/asset_list.dm:145]: what if a subtype overrides it, or double overrides it? or what if someone sets it manually off their new()

Comment (Mothblocks) [reply code/modules/asset_cache/asset_list.dm:145]: As in where you define load_immediately, wrap it in an ifdef

Comment (LemonInTheDark) [reply code/modules/asset_cache/asset_list.dm:145]: no I get that but isn't that still vulnerable to people overriding it to FALSE by accident of OOP, or setting it manually somewhere in New()?

Comment (Mothblocks) [reply code/modules/asset_cache/asset_list.dm:145]: Oh I guess

What you have here feels kinda hacky though, in my head I'm thinking maybe even

```dm
/datum/asset/spritesheet/proc/should_load_immediately()
#ifdef DO_NOT_DEFER_ASSETS
    return TRUE
#else
    return load_immediately
#endif
```

Comment (LemonInTheDark) [reply code/modules/asset_cache/asset_list.dm:145]: hm. ok yeah sure

File: code/modules/asset_cache/asset_list.dm [hunk 2]
```diff
@@ -139,6 +139,13 @@ GLOBAL_LIST_EMPTY(asset_datums)
 	/// Defaults to false so we can process this stuff nicely
 	var/load_immediately = FALSE
 
+/datum/asset/spritesheet/proc/should_load_immediately()
+#ifdef DO_NOT_DEFER_ASSETS
+	return TRUE
+#else
+	return load_immediately
```
Comment (Mothblocks) [inline code/modules/asset_cache/asset_list.dm:146]:
```suggestion
	return load_immediately
#endif
```

**Pull request merged**